charts/k8ssandra/templates/cassandra/superuser-secret.yaml

```diff
@@ -1,11 +1,19 @@
+{{- $password := (include "k8ssandra-common.password" .) }}
+{{- $secret := (lookup "v1" "Secret" .Release.Namespace (include "k8ssandra.superuserSecretName" .)) }}
+
 {{- if and .Values.cassandra.auth.superuser.username (not .Values.cassandra.auth.superuser.secret) .Values.cassandra.auth.enabled }}
+
+{{- if $secret }}
+{{- $password = index $secret.data "password" }}
```

Comment:
This question applies to this, medusa, reaper and stargate.. if I had a username in the Values-file, but changed it - won't my secrets be all messed up in the upgrade process? Currently, only the existing password is fetched in these, but not the existing username. So instead of existing username, we use the one from the values file, but old password. I'll be honest, I have no idea how messed up we could make the system if secret exists, but these are targetting a different username.

Reply:
That is a really good point. I was focused solely on the password. I will update the other secrets so that they reuse the existing username as well.

Reply:
For a first test, I installed with this configuration:

```yaml
cassandra:
  auth:
    enabled: true
  datacenters:
  - name: dc1
    size: 1
reaper:
  enabled: true
stargate:
  enabled: true
  replicas: 1
reaper:
  enabled: true
  cassandraUser:
```
That results in the creation of a secret named `test-reaper` and a Cassandra role/user named `reaper` with password `U59R30EzNxcNUjqi1koK`. I then changed my configuration to:

```yaml
cassandra:
  auth:
    enabled: true
  datacenters:
  - name: dc1
    size: 1
reaper:
  enabled: true
stargate:
  enabled: true
  replicas: 1
reaper:
  enabled: true
  cassandraUser:
    username: reaper-admin
```

After upgrading the username in the secret was changed to `reaper-admin`. The password in the secret did not. A new role was created, `reaper-admin`, same password as for `reaper` role.

I think a really good question is what if anything should happen to the `reaper` role. Should it be deleted in this case? Even if the answer is yes, it would be a post-1.0 change and probably require changes in cass-operator.

There is an additional change that I think should be made. When I change the username, I think a different password should be generated.

Reply:
I think I agree on all fronts there.  I would generally expect that if the username is changed, that should effectively create a new user, not reuse the password from the older account.  I also think the cleanup of the left-over role can wait until post-1.0.0.

Reply:
https://github.com/datastax/cass-operator/issues/382 is related and something with which we need to be involved.

Reply:
I'm fine with postponing these to post-1.0